### ISSUE TITLE
Add/fix `Bad credential` error handling

### ISSUE THREAD

**Issue by phlax** [envoyproxy/pytooling#332]
If you try to run the dependency checker with bad credentials - ie a key that doesnt have permission to access some/all of the API resources required it can throw an error, which currently appears to be unhandled.

**Comment by phlax** [comment]
sample output

```
DependencyChecker NOTICE Running checks: cves, release_dates, release_issues, release_sha, releases
DependencyChecker NOTICE Preloading: cves, release_dates, release_issues, release_sha, releases
DependencyChecker INFO boringssl_fips is not a GitHub repository
https://commondatastorage.googleapis.com/chromium-boringssl-fips/boringssl-ae223d6138807a13006342edfeef32e813246b39.tar.xz
DependencyChecker INFO com_googlesource_chromium_base_trace_event_common is not a GitHub repository
https://storage.googleapis.com/envoyproxy-wee8/chromium-base_trace_event_common-7f36dbc19d31e2aad895c60261ca8f726442bfbb.tar.gz
DependencyChecker INFO com_googlesource_chromium_zlib is not a GitHub repository
https://storage.googleapis.com/envoyproxy-wee8/chromium-zlib-fc5cfd78a357d5bb7735a58f383634faaafe706a.tar.gz
DependencyChecker INFO com_googlesource_googleurl is not a GitHub repository
https://storage.googleapis.com/quiche-envoy-integration/googleurl_561705e0066ff11e6cb97b8092f1547835beeb92.tar.gz
DependencyChecker INFO kafka_server_binary is not a GitHub repository
https://archive.apache.org/dist/kafka/3.1.0/kafka_2.13-3.1.0.tgz
DependencyChecker INFO v8 is not a GitHub repository
https://storage.googleapis.com/envoyproxy-wee8/v8-9.9.115.5.tar.gz
DependencyChecker ERROR [releases] Check disabled: data download (preload_releases) failed Bad credentials
DependencyChecker ERROR [release_issues] Check disabled: data download (preload_releases) failed Bad credentials
DependencyChecker ERROR [release_dates] Check disabled: data download (preload_releases) failed Bad credentials
asyncio: ERROR _GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=BadRequest('Bad credentials')>
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_core/aio/core/functional/decorators.py", line 87, in async_result
    return self.set_prop_cache(instance, await self.fun(instance))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/external/base_pip3_envoy_dependency_check/envoy/dependency/check/abstract/dependency.py", line 152, in recent_commits
    await self.commits_since_current
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_core/aio/core/functional/decorators.py", line 87, in async_result
    return self.set_prop_cache(instance, await self.fun(instance))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/external/base_pip3_envoy_dependency_check/envoy/dependency/check/abstract/dependency.py", line 49, in commits_since_current
    since=await self.release.timestamp_commit).total_count
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_core/aio/core/functional/decorators.py", line 87, in async_result
    return self.set_prop_cache(instance, await self.fun(instance))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/external/base_pip3_envoy_dependency_check/envoy/dependency/check/abstract/release.py", line 143, in timestamp_commit
    return (await self.commit).timestamp
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_core/aio/core/functional/decorators.py", line 87, in async_result
    return self.set_prop_cache(instance, await self.fun(instance))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/external/base_pip3_envoy_dependency_check/envoy/dependency/check/abstract/release.py", line 67, in commit
    raise e
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/external/base_pip3_envoy_dependency_check/envoy/dependency/check/abstract/release.py", line 63, in commit
    return await self.repo.commit(self.tag_name)
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_api_github/aio/api/github/abstract/repo.py", line 45, in commit
    await self.getitem(f"commits/{name}"))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_api_github/aio/api/github/abstract/repo.py", line 58, in getitem
    return await self.github.getitem(self.github_endpoint(query))
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_aio_api_github/aio/api/github/abstract/api.py", line 112, in getitem
    return await self.api.getitem(*args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_gidgethub/gidgethub/abc.py", line 133, in getitem
    data, _ = await self._make_request(
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_gidgethub/gidgethub/abc.py", line 114, in _make_request
    data, self.rate_limit, more = sansio.decipher_response(*response)
  File "/root/.cache/bazel/_bazel_root/bbe9e67c575cc5f19db85a52451b15a6/execroot/envoy/bazel-out/k8-fastbuild/bin/tools/dependency/check.runfiles/base_pip3_gidgethub/gidgethub/sansio.py", line 366, in decipher_response
    raise exc_type(*args)
gidgethub.BadRequest: Bad credentials

```

**Comment by phlax** [comment]
what is confusing is that if you run a check individually the exception seems to be handled

**Comment by phlax** [comment]
as discussed offline the issue is related to `ConcurrentError` handling

**Comment by phlax** [comment]
i think this is fixed by #418
